.Lib9c.Tests/Action/EventMaterialItemCraftsTest.cs

```diff
@@ -0,0 +1,278 @@
+                    [700108] = 5,
+                    [700206] = 5,
+                },
+            };
+        }
+
+        public static IEnumerable<object[]> GetExecuteInvalidAddressExceptionMemberData()
+        {
+            yield return new object[]
+            {
+                1002,
+                10020001,
+                new Dictionary<int, int>
+                {
+                    [700102] = 5,
+                    [700204] = 5,
+                },
+            };
+            yield return new object[]
+            {
+                1002,
+                10020002,
+                new Dictionary<int, int>
+                {
+                    [700108] = 10,
+                    [700206] = 5,
+                },
+            };
+            yield return new object[]
+            {
+                1002,
+                10020002,
+                new Dictionary<int, int>
+                {
+                    [700102] = 10,
+                    [700104] = 5,
+                },
+            };
+            yield return new object[]
+            {
+                1002,
+                10020002,
+                new Dictionary<int, int>
+                {
+                    [700102] = 10,
+                    [700206] = 5,
+                },
+            };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetExecuteSuccessMemberData))]
+        public void Execute_Success(
+            int eventScheduleId,
+            int eventMaterialItemRecipeId,
+            Dictionary<int, int> materialsToUse)
+        {
+            Assert.True(_tableSheets.EventScheduleSheet
+                .TryGetValue(eventScheduleId, out var scheduleRow));
+            var contextBlockIndex = scheduleRow.StartBlockIndex;
+            Execute(
+                _initialStates,
+                eventScheduleId,
+                eventMaterialItemRecipeId,
+                materialsToUse,
+                contextBlockIndex);
+            contextBlockIndex = scheduleRow.RecipeEndBlockIndex;
+            Execute(
+                _initialStates,
+                eventScheduleId,
+                eventMaterialItemRecipeId,
+                materialsToUse,
+                contextBlockIndex);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetExecuteInvalidAddressExceptionMemberData))]
+        public void Execute_InvalidAddressException(
+            int eventScheduleId,
+            int eventMaterialItemRecipeId,
+            Dictionary<int, int> materialsToUse)
+        {
+            Assert.True(_tableSheets.EventScheduleSheet
+                .TryGetValue(1002, out var scheduleRow));
+            Assert.Throws<InvalidMaterialCountException>(() =>
+            {
+                Execute(
+                    _initialStates,
+                    eventScheduleId,
+                    eventMaterialItemRecipeId,
+                    materialsToUse,
+                    scheduleRow.StartBlockIndex);
+            });
+        }
```

Comment:
The name is `Execute_InvalidAddressException`, but why do you check `InvalidMaterialCountException`?

Reply:
Oh...........that's a typo